src/utils.ts

```diff
@@ -84,15 +84,15 @@ export function hasTrailingSlash(
   respectQueryAndFragment?: boolean
 ): boolean {
   if (!respectQueryAndFragment) {
-    return input.endsWith("/");
+    return !input.includes("?") && input.endsWith("/");
```

Comment:
(notice that by default, we do not handle `query`. we probably don't want to change logic in this line)

Reply:
i see that query is not handled by default, but what to do with these cases? 

because, currently, all these calls return `true`:

```ts
hasTrailingSlash('/a?b=/', false)
hasTrailingSlash('/a/?b=/', false)
hasTrailingSlash('/a?b=/', true)
hasTrailingSlash('/a/?b=/', true)
```

Reply:
Checking, without this change, only failing test is `withoutTrailingSlash('foo/?k=/', false /* no respect */)`

Reply:
I have revert it for now since change on `hasProcol` like this is risky to introduce behavior changes.